### PR TITLE
Added types for crypto-ld

### PR DESCRIPTION
**Pull request by behruzrahimov**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by pleerock — APPROVED** [review]

File: types/crypto-ld/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Type definitions for crypto-ld 7.0
+// Project: https://github.com/digitalbazaar/crypto-ld
+// Definitions by: behruzrahimov <https://github.com/behruzrahimov>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export class CryptoLD {
+    constructor(params?: { suites: any });
```
Comment (peterblazejewicz) [inline types/crypto-ld/index.d.ts:7]: this looks to be a Map, keyed by string, as per comments in source, not `any`, at least `Map<string, unknown>` IMO

File: types/crypto-ld/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    _installed({ type }: { type: string }): boolean;
+
+    _suiteForType(document: object): object;
+}
+
+export class LDKeyPair {
+    constructor({ id, controller, revoked }?: { id?: string; controller?: string; revoked?: string });
+
+    static generate(/* options */): Promise<LDKeyPair>;
```
Comment (peterblazejewicz) [inline types/crypto-ld/index.d.ts:45]: do you want to expose this one?
looks like a hook for extensibility only:
https://github.com/digitalbazaar/crypto-ld/blob/main/lib/LDKeyPair.js#L55

File: types/crypto-ld/index.d.ts (hunk 1 above)
Comment (behruzrahimov) [reply types/crypto-ld/index.d.ts]: fixed

File: types/crypto-ld/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }: {
+        id: string;
+        documentLoader: () => void;
+        checkContext: boolean;
+        checkRevoked: boolean;
+    }): Promise<LDKeyPair>;
+
+    _installed({ type }: { type: string }): boolean;
+
+    _suiteForType(document: object): object;
```
[the remark this answers is not in the document]
Comment (behruzrahimov) [reply types/crypto-ld/index.d.ts]: fixed

**Review by peterblazejewicz — COMMENTED** [review]
LGTM!
@behruzrahimov thanks!

**Review by pleerock — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by behruzrahimov** [comment]
Ready to merge

**Pull request merged**